File: GXpert/GXpert.Web/Modules/Attendance/AssignmentAttemptEvaluation/AssignmentAttemptEvaluationColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Attendance.Columns;

[ColumnsScript("Attendance.AssignmentAttemptEvaluation")]
[BasedOnRow(typeof(AssignmentAttemptEvaluationRow), CheckNames = true)]
public class AssignmentAttemptEvaluationColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string AssignmentTitle { get; set; }
    public string AssignmentAttemptFileUploaded { get; set; }
    public string AssignmentEvaluationEvalutionCriteria { get; set; }
    public int MarksObtained { get; set; }
    public short EvaluationStatus { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Attendance/AssignmentAttemptEvaluationColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { AssignmentAttemptEvaluationRow } from "./AssignmentAttemptEvaluationRow";

export interface AssignmentAttemptEvaluationColumns {
    Id: Column<AssignmentAttemptEvaluationRow>;
    AssignmentTitle: Column<AssignmentAttemptEvaluationRow>;
    AssignmentAttemptFileUploaded: Column<AssignmentAttemptEvaluationRow>;
    AssignmentEvaluationEvalutionCriteria: Column<AssignmentAttemptEvaluationRow>;
    MarksObtained: Column<AssignmentAttemptEvaluationRow>;
    EvaluationStatus: Column<AssignmentAttemptEvaluationRow>;
    InsertDate: Column<AssignmentAttemptEvaluationRow>;
    InsertUserId: Column<AssignmentAttemptEvaluationRow>;
    UpdateDate: Column<AssignmentAttemptEvaluationRow>;
    UpdateUserId: Column<AssignmentAttemptEvaluationRow>;
    IsActive: Column<AssignmentAttemptEvaluationRow>;
}

export class AssignmentAttemptEvaluationColumns extends ColumnsBase<AssignmentAttemptEvaluationRow> {
    static readonly columnsKey = 'Attendance.AssignmentAttemptEvaluation';
    static readonly Fields = fieldsProxy<AssignmentAttemptEvaluationColumns>();
}